src/processors/pinContentToIpfs.ts

```diff
@@ -0,0 +1,80 @@
+import * as IPFS from "ipfs-http-client";
+import createProcessor from "../processor";
+
+export type FileContent = Uint8Array;
+
+export class Pinner {
+  ipfsOptions: IPFS.Options;
+  ipfs: IPFS.IPFSHTTPClient;
+  where: string;
```

Comment:
A lot of refactoring here, but I believe i preserved all the existing logic.
This is worth a solid read through.